markdown-to-slack-blocks-tool/server.js

```diff
@@ -0,0 +1,18 @@
+const mack = require('@tryfabric/mack');
+
+async function run() {
+  let blocks = await mack.markdownToBlocks(process.env.INPUT.replace("\\n", "\n"));
+  blocks = blocks.map(b => {
+    if (b.text && b.text.text) {
+      b.text.text = b.text.text
+          .replace(/&lt;@([A-Za-z0-9]+)&gt;/g, "<@$1>") // User mentions
+          .replace(/&lt;!([A-Za-z0-9]+\^[A-Za-z0-9]+)&gt;/g, "<!$1>") // Group mentions
+          .replace(/&lt;!([A-Za-z0-9\|]+)&gt;/g, "<!$1>") // Special mentions
+          .replace(/&lt;#([A-Za-z0-9]+)&gt;/g, "<#$1>"); // Channels
+    }
+    return b;
+  })
+  console.log(JSON.stringify(blocks));
+}
```

Comment:
```suggestion
async function run() {
  let input = process.env.INPUT
      .replace("\\n", "\n")
      .replace(/<@([A-Za-z0-9]+)>/g, "&lt;@$1&gt;") // User mentions
      .replace(/<!([A-Za-z0-9\|\^]+)>/g, "&lt;!$1&gt;") // Group / Special mentions
      .replace(/<#([A-Za-z0-9]+)>/g, "&lt;#$1&gt;") // Channels
    
  let blocks = await mack.markdownToBlocks(input);
  blocks = blocks.map(b => {
    if (b.text && b.text.text) {
      b.text.text = b.text.text
          .replace(/&lt;@([A-Za-z0-9]+)&gt;/g, "<@$1>") // User mentions
          .replace(/&lt;!([A-Za-z0-9\|\^]+)&gt;/g, "<!$1>") // Group / Special mentions
          .replace(/&lt;#([A-Za-z0-9]+)&gt;/g, "<#$1>") // Channels
    }
    return b;
  })
  console.log(JSON.stringify(blocks)); 
}
```

I forget _why_ but this is the latest of the code I was using. There was some issue about links or mentions or something, right?

Reply:
This would have been the mentions. I ended up changing my mentions to do the input shaping but I think it's good to do it here so others don't have to do the hack.